packages/atomic/cypress/integration/facets/category-facet.cypress.ts

```diff
@@ -0,0 +1,456 @@
+        cy.get('@categoryClearAllButton').click();
+        cy.wait('@coveoAnalytics');
+      });
+
+      it('Should go back to the first parent level', () => {
+        cy.get('@categoryClearAllButton').should('not.exist');
+        assertNonZeroFacetCount();
+      });
+
+      it('Should log UA correctly', () => {
+        cy.wait('@coveoAnalytics').then((intercept: any) => {
+          const analyticsBody = intercept.request.body;
+          expect(analyticsBody).to.have.property(
+            'actionCause',
+            'facetClearAll'
+          );
+          expect(analyticsBody.customData).to.have.property(
+            'facetField',
+            categoryFacetProp.field
+          );
+        });
+      });
+    });
+
+    describe('When user clicks ShowMore button', () => {
+      it('Should expand list of childrens on same level', () => {
+        assertClickShowMore(categoryFacetProp.numberOfValue);
+      });
+
+      it('Should log UA correctly', () => {
+        cy.wait('@coveoAnalytics');
+        assertShowMoreUA(categoryFacetProp.field);
+      });
+
+      describe('When user click ShowLess button', () => {
+        it('Should collapse the list and hide ShowLess button', () => {
+          assertClickShowLess(categoryFacetProp.numberOfValue * 2);
+        });
+
+        it('Should log UA correctly', () => {
+          cy.wait('@coveoAnalytics');
+          assertShowLessUA(categoryFacetProp.field);
+        });
+      });
+    });
+  });
+
+  describe('When user goes to last child on level 3', () => {
+    beforeEach(() => {
+      clickOnCategoryFacetWithValue(canadaHierarchy[0]);
+      cy.wait('@coveoAnalytics');
+      clickOnCategoryFacetWithValue(canadaHierarchy[1]);
+      cy.wait('@coveoAnalytics');
+      clickOnCategoryFacetWithValue(canadaHierarchy[2]);
+      cy.wait(500);
+    });
+
+    it('Should have correct 3 parents level up and default Category Label', () => {
+      assertClearAllTitleAndTotalParents(
+        categoryFacetProp.defaultCategoryLabel,
+        3
+      );
+    });
+
+    it.skip('Should log UA correctly', () => {
+      //TODO: Enable it when UA log facetValue correctly
+      cy.wait('@coveoAnalytics').then((intercept: any) => {
+        const analyticsBody = intercept.request.body;
+        expect(analyticsBody).to.have.property('actionCause', 'facetSelect');
+        expect(analyticsBody.customData).to.have.property(
+          'facetField',
+          categoryFacetProp.field
+        );
+        expect(analyticsBody.customData).to.have.property(
+          'facetValue',
+          convertArrayToString(canadaHierarchy)
+        );
+      });
+    });
+
+    it('Should not contain any other level', () => {
+      cy.get(FacetAlias.facetShadow)
+        .find('ul[part="parents"]')
+        .find('li')
+        .contains(canadaHierarchy[2])
+        .find('div.arrow-size')
+        .should('not.exist');
+    });
+
+    it('Should display correctly in URL hash', () => {
```

Comment:
Is there a test where you load a selected path in the url to see if the right value is selected in the facet?
And maybe a test where the hash is modified to see if the state is updated correctly?

Reply:
right now only Pager has this check, I haven't added these type of tests for all facets yet, I will add that